There is a build issue:

```
[CXX] src\library\parser.cpp
parser.cpp
src\library\parser.cpp(71): error C2601: 'Parser::isUtf8': local function definitions are illegal
src\library\parser.cpp(40): note: this line contains a '{' which has not yet been matched
src\library\parser.cpp(149): error C2601: 'Parser::playlistEntrytoLocalFile': local function definitions are illegal
src\library\parser.cpp(40): note: this line contains a '{' which has not yet been matched
src\library\parser.cpp(40): fatal error C1075: the left brace '{' was unmatched at the end of the file
scons: *** [win64_build\library\parser.obj] Error 2
scons: building terminated because of errors.
```

--- a/src/library/parserm3u.cpp
+++ b/src/library/parserm3u.cpp
@@ -97,8 +97,10 @@ QList<QString> ParserM3u::parse(QString sFilename)
 QString ParserM3u::getFilepath(QTextStream* stream, QString basepath) {
     QString textline;
     while (!(textline = stream->readLine().trimmed()).isEmpty()) {
-        if (textline.startsWith("#")) {
-            // Skip comments
+        if (textline.contains("#")) {
+            // Skip all lines with comments or special tags that are
+            // prefixed with the reserved character '#'. In file URLs
+            // this special character needs to be encoded as "%23".
             continue;
         }
         QString trackLocation = playlistEntrytoLocalFile(textline);

--- a/src/library/parserm3u.cpp
+++ b/src/library/parserm3u.cpp
@@ -47,84 +47,75 @@ ParserM3u::~ParserM3u()
 
 }
 
-
 QList<QString> ParserM3u::parse(QString sFilename)
 {
-    QFile file(sFilename);
-    QString basepath = sFilename.section('/', 0, -2);
-
     clearLocations();
-    //qDebug() << "ParserM3u: Starting to parse.";
-    if (file.open(QIODevice::ReadOnly) && !isBinary(sFilename)) {
-        /* Unfortunately, QT 4.7 does not handle <CR> (=\r or asci value 13) line breaks.
-         * This is important on OS X where iTunes, e.g., exports M3U playlists using <CR>
-         * rather that <LF>
-         *
-         * Using QFile::readAll() we obtain the complete content of the playlist as a ByteArray.
-         * We replace any '\r' with '\n' if applicaple
-         * This ensures that playlists from iTunes on OS X can be parsed
-         */
-        QByteArray ba = file.readAll();
-        //detect encoding
-        bool isCRLF_encoded = ba.contains("\r\n");
-        bool isCR_encoded = ba.contains("\r");
-        if(isCR_encoded && !isCRLF_encoded)
-            ba.replace('\r','\n');
-        QTextStream textstream(ba.constData());
 
-        if (isUtf8(ba.constData())) {
-            textstream.setCodec("UTF-8");
-        } else {
-            textstream.setCodec("windows-1252");
-        }
-
-        while (!textstream.atEnd()) {
-            QString sLine = getFilepath(&textstream, basepath);
-            if (sLine.isEmpty()) {
-                continue;
-            }
-            //qDebug() << "ParserM3u: parsed: " << (sLine);
-            m_sLocations.append(sLine);
-        }
-
-        file.close();
+    QFile file(sFilename);
+    if (isBinary(sFilename) || !file.open(QIODevice::ReadOnly)) {
+        qWarning()
+                << "Failed to open playlist file"
+                << sFilename;
         return m_sLocations;
     }
 
-    file.close();
-    return QList<QString>(); //if we get here something went wrong
+    // Unfortunately QTextStream does not handle <CR> (=\r or asci value 13) line breaks.
+    // This is important on OS X where iTunes, e.g., exports M3U playlists using <CR>
+    // rather that <LF>.
+    //
+    // Using QFile::readAll() we obtain the complete content of the playlist as a ByteArray.
+    // We replace any '\r' with '\n' if applicaple
+    // This ensures that playlists from iTunes on OS X can be parsed
+    QByteArray ba = file.readAll();
+    //detect encoding
+    bool isCRLF_encoded = ba.contains("\r\n");
+    bool isCR_encoded = ba.contains("\r");
+    if (isCR_encoded && !isCRLF_encoded) {
+        ba.replace('\r', '\n');
+    }
+
+    QTextStream textstream(ba.constData());
+    if (isUtf8(ba.constData())) {
+        textstream.setCodec("UTF-8");
+    } else {
+        textstream.setCodec("windows-1252");
+    }
+
+    const QString basepath = sFilename.section('/', 0, -2);
+    while (!textstream.atEnd()) {
+        QString sLine = getFilepath(&textstream, basepath);
+        if (sLine.isEmpty()) {
+            continue;
+        }
+        m_sLocations.append(sLine);
+    }
+
+    return m_sLocations;
 }
 
 
 QString ParserM3u::getFilepath(QTextStream* stream, QString basepath) {
-    QString textline = stream->readLine();
-
-    while (!textline.isEmpty()) {
-        //qDebug() << "Untransofrmed text: " << textline;
-        if (textline.isNull()) {
-            break;
+    QString textline;
+    while (!(textline = stream->readLine().trimmed()).isEmpty()) {
+        if (textline.startsWith("#")) {
+            // Skip comments
+            continue;
         }
-
-        if (!textline.contains("#")) {
-            QString trackLocation = playlistEntrytoLocalFile(textline);
-            if(QFile::exists(trackLocation)) {
-                return trackLocation;
-            } else {
-                // Try relative to m3u dir
-                QString rel = QDir(basepath).filePath(trackLocation);
-                if (QFile::exists(rel)) {
-                    return rel;
-                }
-                // We couldn't match this to a real file so ignore it
-                qWarning() << trackLocation << "not found";
+        QString trackLocation = playlistEntrytoLocalFile(textline);
+        if (QFileInfo::exists(trackLocation)) {
+            return trackLocation;
+        } else {
+            // Try relative to m3u dir
+            QString rel = QDir(basepath).filePath(trackLocation);
+            if (QFile::exists(rel)) {
+                return rel;
             }
+            // We couldn't match this to a real file so ignore it
+            qWarning() << trackLocation << "not found";
         }
-        textline = stream->readLine();
     }
-
     // Signal we reached the end
-    return 0;
-
+    return QString();
 }
 
 bool ParserM3u::writeM3UFile(const QString &file_str, QList<QString> &items, bool useRelativePath) {

--- a/src/library/parserm3u.cpp
+++ b/src/library/parserm3u.cpp
@@ -97,10 +97,8 @@ QList<QString> ParserM3u::parse(QString sFilename)
 QString ParserM3u::getFilepath(QTextStream* stream, QString basepath) {
     QString textline;
     while (!(textline = stream->readLine().trimmed()).isEmpty()) {
-        if (textline.contains("#")) {
-            // Skip all lines with comments or special tags that are
-            // prefixed with the reserved character '#'. In file URLs
-            // this special character needs to be encoded as "%23".
+        if (textline.startsWith("#")) {
+            // Skip comments
             continue;
         }
         QString trackLocation = playlistEntrytoLocalFile(textline);

